application/client/src/app/ui/tabs/sources/multiplefiles/component.ts

```diff
@@ -0,0 +1,115 @@
+import { Component, AfterContentInit, Input, HostListener } from '@angular/core';
+import { Ilc, IlcInterface } from '@env/decorators/component';
+import { Initial } from '@env/decorators/initial';
+import { FileType, File } from '@platform/types/files';
+import { Action } from '../common/actions/action';
+import { FileHolder } from './file.holder';
+import { bytesToStr } from '@env/str';
+import { Holder } from '@module/matcher';
+
+@Component({
+    selector: 'app-tabs-source-multiple-files',
+    templateUrl: './template.html',
+    styleUrls: ['./styles.less'],
+})
+@Initial()
+@Ilc()
+export class TabSourceMultipleFiles extends Holder implements AfterContentInit {
+    @Input() files!: File[];
+
+    @HostListener('window:keydown', ['$event'])
+    handleKeyDown(event: KeyboardEvent) {
+        this.matcher.search(event.key);
+        this.ngFiles = this._files
+            .sort((a: FileHolder, b: FileHolder) => b.getScore() - a.getScore())
+            .filter((file: FileHolder) => file.getScore() > 0);
+    }
+
+    public action: Action = new Action();
+    public ngCompatible: boolean = false;
+    public ngSelectedCount: number = 0;
+    public ngSelectedSize: string = '0 b';
+    public ngDestination: string = '';
+    public ngFiles: FileHolder[] = [];
+
+    private _files: FileHolder[] = [];
+
+    constructor() {
+        super();
+    }
+
+    public ngAfterContentInit(): void {
+        this._files = this.files.map((file: File) => new FileHolder(this.matcher, file));
+        this.ngFiles = this._files;
+        this.ngOnContext(null);
```

Comment:
good place to restore state of your view (for example if it was opened, but user "jumped" to another tab and back
in `ngOnDestroy` you can save state. See API bellow in comments